python/cuspatial/cuspatial/_lib/shapefile_reader.pyx

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+
+from cudf.core.column import Column
+from cudf._lib.cudf import *
+from libc.stdlib cimport calloc, malloc, free
+from libcpp.pair cimport pair
+
+cpdef cpp_read_polygon_shapefile(shapefile_file_name):
+    cdef bytes py_bytes = shapefile_file_name.encode()
+    cdef char* c_string = py_bytes
+    cdef gdf_column* c_ply_fpos = <gdf_column*>malloc(sizeof(gdf_column))
+    cdef gdf_column* c_ply_rpos = <gdf_column*>malloc(sizeof(gdf_column))
+    cdef gdf_column* c_ply_x = <gdf_column*>malloc(sizeof(gdf_column))
+    cdef gdf_column* c_ply_y = <gdf_column*>malloc(sizeof(gdf_column))
```

Comment:
These 4 are all leaking as far as I can tell. I don't believe `gdf_column_to_column_mem` frees the memory.

Reply:
Thanks, that was careless. There are other areas I need to double check here now.